### ISSUE TITLE
Objective / Unambiguous metadata in definitions?

### ISSUE THREAD

**Issue by WilcoFiers** [act-rules/act-rules.github.io#1003]
I've seen a few places where a subjective or sometimes even ambiguous definition is used in some of our rules. I'm wondering if, to ensure that reviewers look at this, and to prevent us from using definitions we shouldn't be using, we should add the some frontmatter info to all the definitions.

Example:

```
---
title: Focusable
key: focusable
objective: true
unambiguous: true
---

Elements that can become the target of keyboard input as described in the [HTML](https://www.w3.org/TR/html) specification of [focusable](https://html.spec.whatwg.org/#focusable-area) and [can be focused](https://html.spec.whatwg.org/#specially-focusable).
```

If nothing else, I'm hoping this can maybe remind reviewers to check that these things are correct.

**Comment by Jym77** [comment]
I like that. It should even be possible (although somewhat tricky) to automatically figure out where the definition is used (applicability, expectation, example description, note, assumption, …) and test if this is allowed with the current "ambiguity level".

Could also be a single "value" instead of several booleans. It looks like there is a gradation "ambiguous < subjective < objective" as far as I understand.

**Comment by WilcoFiers** [comment]
Objective and ambiguous are two distinct things. The distinction is important in rules, because in ACT, applicability has to be objective, whereas the expectation can be subjective. For example "descriptive" is a subjective term. In the expectation you can say that the alt text has to be descriptive of the image, but you can't write an applicability that says all images with a descriptive alt text.

Ambiguous also matters, sometimes we have ambiguous definitions in assumptions, to reference concepts from WCAG, things like "essential".

**Comment by Jym77** [comment]
I agree that objective/subjective and (un)ambiguous are different things.

What I fail to see is how it's possible to have something which is both objective and ambiguous.

---
As far as I understand,
* "objective" means "all users will always interpret that the same way".
* "subjective" means "two different users may interpret that in different way".
* "ambiguous" means "a given user may interpret that in different ways depending on other stuff" (possibly, user's mood).

Maybe I'm wrong with these understandings…

Thus, I fail to find something that could be objective (you and me will always interpret it the same way) but ambiguous (I might interpret it in different ways, you might interpret is in different ways). Which is why I feel there is a gradation rather than two orthogonal dimensions.

This is a completely minor point and I'm fine with two booleans.

**Comment by Jym77** [comment]
ACT rules format states under [Applicability](https://www.w3.org/TR/act-rules-format/#applicability):
> When used in applicability, potentially ambiguous concepts must be defined objectively.

The fact that **ambiguous** concepts must be defined **objectively** (and not **unambiguously**) support the interpretation that "objective and ambiguous" is impossible and that even if these are different things they are not independant and a tri-valued field can encode both ("ambiguous/subjective/objective" are the only three possibilities).

**Comment by EmmaJP** [comment]
Seems sensible ... need to add something to rule guidance

**Comment by Jym77** [comment]
If we want to label some WCAG defs as "ambiguous", I think we should also come up with examples of the ambiguity ("does this mean A or B?", ...)  to strengthen our case and not look like we're blindly throwing names...

**Comment by EmmaJP** [comment]
Shadi also recommended feeding back to WCAG 2.2 AG so that any potential changes can be actioned as soon as possible.

**Comment by EmmaJP** [comment]
"objective" - not influenced by a persons feelings or opinions (is there an alt attribute?)
"subjective" - influenced by personal feelings, tastes or opinions (is the alt attribute appropriate?)
"ambiguous" - can be interpreted in more than one way (does the page have a title? `<title>` or `<h1>`)
"unambiguous" - not open to more than one interpretation (does the page have a `<title>`?)

**Comment by Jym77** [comment]
@EmmaJP Thanks. Good examples.
Probably need to add something like that in the design document 😉

**Comment by patrickhlauke** [comment]
some concept of "requiring human judgement" could be inserted in there somewhere (in the objective/subjective definition) - though yes, human judgement will then be influenced by feelings/tastes/opinions, but also a human's ability to understand context, human language processing, experience...so not completely down to "whether they like it or not"

**Comment by Jym77** [comment]
First part of the discussion: https://www.w3.org/2019/11/28-act-r-minutes.html#item04

**Comment by EmmaJP** [comment]
Any guidance on this should also cover where different definitions can be used in respect of these things.

**Comment by Jym77** [comment]
Closed by #1085.
